diff --git a/BuildLinux.sh b/BuildLinux.sh
old mode 100644
new mode 100755
index 1d673a7..f4f957f
--- a/BuildLinux.sh
+++ b/BuildLinux.sh
@@ -16,7 +16,7 @@ fi
 set -e # exit on first error
 
 function check_available_memory_and_disk() {
-    FREE_MEM_GB=$(free -g -t | grep 'Mem' | rev | cut -d" " -f1 | rev)
+    FREE_MEM_GB=$(free -g | awk '/^Mem:/{mem=$NF} /^Swap:/{swap=$NF} END{print mem+swap}')
     MIN_MEM_GB=10
 
     FREE_DISK_KB=$(df -k . | tail -1 | awk '{print $4}')
diff --git a/DockerBuild.sh b/DockerBuild.sh
old mode 100644
new mode 100755
diff --git a/DockerRun.sh b/DockerRun.sh
old mode 100644
new mode 100755
diff --git a/build_release_macos.sh b/build_release_macos.sh
old mode 100644
new mode 100755
diff --git a/deps/MPFR/MPFR.cmake b/deps/MPFR/MPFR.cmake
index 1161a1c..6e525eb 100644
--- a/deps/MPFR/MPFR.cmake
+++ b/deps/MPFR/MPFR.cmake
@@ -25,8 +25,8 @@ else ()
     endif ()
 
     ExternalProject_Add(dep_MPFR
-        URL https://www.mpfr.org/mpfr-current/mpfr-4.2.1.tar.bz2
-        URL_HASH SHA256=b9df93635b20e4089c29623b19420c4ac848a1b29df1cfd59f26cab0d2666aa0
+        URL https://www.mpfr.org/mpfr-current/mpfr-4.2.2.tar.bz2
+        URL_HASH SHA256=9ad62c7dc910303cd384ff8f1f4767a655124980bb6d8650fe62c815a231bb7b
         DOWNLOAD_DIR ${DEP_DOWNLOAD_DIR}/MPFR
         BUILD_IN_SOURCE ON
         CONFIGURE_COMMAND autoreconf -f -i && 
diff --git a/flatpak/io.github.softfever.OrcaSlicer.yml b/flatpak/io.github.softfever.OrcaSlicer.yml
index f5a148c..99f1b6c 100644
--- a/flatpak/io.github.softfever.OrcaSlicer.yml
+++ b/flatpak/io.github.softfever.OrcaSlicer.yml
@@ -214,9 +214,9 @@ modules:
 
       # MPFR
       - type: file
-        url: https://www.mpfr.org/mpfr-current/mpfr-4.2.1.tar.bz2
+        url: https://www.mpfr.org/mpfr-current/mpfr-4.2.2.tar.bz2
         dest: external-packages/MPFR
-        sha256: b9df93635b20e4089c29623b19420c4ac848a1b29df1cfd59f26cab0d2666aa0
+        sha256: 9ad62c7dc910303cd384ff8f1f4767a655124980bb6d8650fe62c815a231bb7b
 
       # NLopt
       - type: file
diff --git a/linux.d/debian b/linux.d/debian
index fd13c70..a98d490 100644
--- a/linux.d/debian
+++ b/linux.d/debian
@@ -20,7 +20,7 @@ REQUIRED_DEV_PACKAGES=(
     libssl-dev
     libtool
     libudev-dev
-    libwebkit2gtk-4.0-dev
+    libwebkit2gtk-4.1-dev
     ninja-build
     texinfo
     wget
diff --git a/run_gettext.sh b/run_gettext.sh
old mode 100644
new mode 100755
