diff --git a/BuildLinux.sh b/BuildLinux.sh
old mode 100644
new mode 100755
index 1d673a71..67de789a
--- a/BuildLinux.sh
+++ b/BuildLinux.sh
@@ -16,7 +16,8 @@ fi
 set -e # exit on first error
 
 function check_available_memory_and_disk() {
-    FREE_MEM_GB=$(free -g -t | grep 'Mem' | rev | cut -d" " -f1 | rev)
+    FREE_MEM_GB=$(free -g | awk '/^Mem:/{mem=$NF} /^Swap:/{swap=$NF} END{print mem+swap}')
     MIN_MEM_GB=10
 
     FREE_DISK_KB=$(df -k . | tail -1 | awk '{print $4}')
